### PR TITLE
[WIP] Vertical Zoom

### PR DESCRIPTION
**Pull request by Spekular**
I decided it was about time I took #3763 into my own hands. My very, *very*, incompetent hands.

Right now, shift scroll:

1. Changes a zoom-selector to the right of the horizontal scroll selector. This works fine except that it also changes the song editor to select mode while shift is held, and the selectors need padding and labels/icons.
2. Triggers a big mess that tries to set all tracks to be 300 units tall. This needs to actually set height properly based on the track's base height and the song editor's zoom level.

The big mess goes through a bunch of classes that it maybe shouldn't because I can't wrap my brain around how everything is connected. The tracks don't actually visually change size until you save and reload the project, and BB tracks are unaffected because why not.

<sup>send help</sup>

### CODE REVIEW

**Comment by qnebra** [comment]
At this moment, vertical zooming not works.

**Comment by Spekular** [comment]
@qnebra yeah, this isn't actually working yet. I wouldn't recommend testing it with any actual projects either, since it sets track heights absurdly high.

**Comment by qnebra** [comment]
Maybe limit vertical zoom to max 400%?

File: plugins/LadspaEffect/caps/basics.h [hunk 1]
```diff
@@ -184,7 +184,7 @@ class Plugin {
 		inline sample_t getport_unclamped (int i)
 			{
 				sample_t v = *ports[i];
-				return (isinf (v) || isnan(v)) ? 0 : v;
+				return (__builtin_isinf_sign(v) || __builtin_isnan(v)) ? 0 : v;
```
Comment (Spekular) [inline plugins/LadspaEffect/caps/basics.h]: Unrelated change required for me to compile.

File: src/gui/editors/SongEditor.cpp [hunk 2]
```diff
@@ -50,6 +50,7 @@
 #include "TimeDisplayWidget.h"
 #include "AudioDevice.h"
 #include "PianoRoll.h"
+#include "Track.h"
```
Comment (Spekular) [inline src/gui/editors/SongEditor.cpp]: Don't actually know if this is necessary

**Comment by Spekular** [comment]
Closing. Issues:
- [ ] Relies on changing the actual size of the tracks rather than changing how they're drawn
- [ ] Doesn't affect BB tracks
- [ ] Doesn't show up until the project is saved and reloaded
- [ ] Sets all tracks to a fixed size instead of actually using the zoom factor
